packages/babel-plugin-transform-template-literals/src/index.js

```diff
@@ -83,19 +83,22 @@ export default declare((api, options) => {
           callExpressionInput.push(t.arrayExpression(raws));
         }
 
-        const init = t.callExpression(helperId, callExpressionInput);
-        annotateAsPure(init);
-        init._compact = true;
-        scope.push({
-          id: templateObject,
-          init,
-          // This ensures that we don't fail if not using function expression helpers
-          _blockHoist: 1.9,
-        });
+        const callExpression = t.callExpression(helperId, callExpressionInput);
+        annotateAsPure(callExpression);
+        callExpression._compact = true;
 
+        const lazyLoad = template.ast`
+          function ${templateObject}() {
+            const data = ${callExpression};
+            ${templateObject} = function() { return data };
+            return data;
+          } 
+        `;
+
+        scope.path.unshiftContainer("body", lazyLoad);
```

Comment:
Actually, do we want to push into the scope and using hoisting?

Reply:
I think unshift is fine, since we're usually pretty consistent about helpers being at the top of each file, and this is helper-like.